### PR TITLE
docs: include guidance on CSS/assets

### LINKED ISSUE #33141: docs: Creating Libraries Guide should include guidance on CSS/assets
# 📚 Docs or angular.io bug report

### Description
I spent some time in January working on a new library. One thing that is missing from the new Library guide, and not supported out of the box (but generally required by most library authors) in the Angular CLI or ng-packagr, is shipping assets or theme SCSS files with your library. This issue (https://github.com/angular/angular-cli/issues/11071) and many other linked/closed issues with hundreds of 👍  indicate that this is a major need and at minimum that the docs to work around this are not available.

It appears that the general guidance for SCSS is to use the `scss-bundle` package similar to [what Angular Material does](https://github.com/angular/components/blob/aaa656dbab41fee298244cc25351da8f44e9f614/tools/BUILD.bazel#L16-L24). For assets, it appears that manually copying them via npm scripts is the most common approach.

I can understand some hesitation to include details about library-specific implementation that could cause some kind of support burden.

However, the other perspective is that library authors see the docs and get started and then get completely stuck on assets and scss files (like many hundreds have already run into), etc with no references for how to proceed. They will then continue to open duplicate issues in the Angular CLI GitHub.

Perhaps there is some kind of happy medium between providing them with some starting points on how to handle this without providing them a fully supported solution that doesn’t exist yet. Even telling/reminding people that you can just use regular command line copy commands to move assets via NPM scripts would help people rather than them having to dig through a lot of closed GitHub issues to find examples that work.

### What's the affected URL?**
- https://next.angular.io/guide/creating-libraries
- https://angular.io/guide/creating-libraries

@jbogarthyde I haven't had time to write up some proposed copy for this, but I wanted to at least create the issue.

### PR DESCRIPTION
**Pull request by santoshyadavdev**
Fixes #33141

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #33141 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by StephenFluin** [comment]
The guide doesn't seem to exist. https://next.angular.io/guide/library-asset-guide

Am I missing something?

**Comment by santoshyadav198613** [comment]
Hi @StephenFluin ,
Please wait this is 

> The guide doesn't seem to exist. https://next.angular.io/guide/library-asset-guide
> 
> Am I missing something?

Hi @StephenFluin , Sorry it was an empty commit I have added some content and discuss now.

**Comment by santoshyadav198613** [comment]
Cc : @jbogarthyde @Splaktar @kapunahelewong this is draft let me know if any changes needed.

File: aio/content/guide/library-asset-guide.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Generating CSS with Angular Libraries
+
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: Perhaps "Including assets with libraries"

File: aio/content/guide/library-asset-guide.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+With Angular Libraries its also possible you may need to process CSS/SCSS/SASS and provide it's output as CSS files, like Angular Material. 
+There are multiple ways to achieve the same, Angular Material uses `scss-bundle` to generate multiple CSS files.
+
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: As a library author, you can process style files in any of the supported formats (CSS, SCSS, or SASS) and generate CSS files to be included with your library. 
There are various ways to do this. Angular Material, for example, uses a utility called [scss-bundle](https://www.npmjs.com/package/scss-bundle)  to generate multiple CSS files.

**Comment by kapunahelewong** [comment]
Thanks for this, @santoshyadav198613. I've requested @jbogarthyde to review as she has a great eye and will have insights on content/structure. For new guides, you also need to add it to the `CODEOWNERS` file. This looks like the right spot: https://github.com/angular/angular/blob/master/.github/CODEOWNERS#L914
@gkalpak, could you confirm that I'm right about the `CODEOWNERS` location?

However, on the point of a new doc, can this info be added to one of the existing libraries docs instead?

File: aio/content/guide/library-asset-guide.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## Using scss-bundle
+
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: This doesn't need to be a separate section.

File: aio/content/guide/library-asset-guide.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+`scss-bundle` is easy to use and easy to configure utility to generate the styles.
+
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: When you have loaded the npm package,  create a configuration file named `scss-bundle.config.json`. 
The configuration tells the tool where to get the bundle and where to put the generated CSS, and lets you provide options for how to generate the styles. The options in this file can be overridden by similar options in the CLI configuration file, `angular.json`. For details, see the [package documentation](https://www.npmjs.com/package/scss-bundle#configuration). 

(now you don't need the rest of this, down to the script)

File: aio/content/guide/library-asset-guide.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+once provided we can add below script in scripts:
+
+```sh
+"build-css" : "scss-bundle -p <project-path-to-scss-bundle.config.json>"
+```
+
+## Copying assets using custom builder
+
+There is a custom builder available to copy the assets folder `@linnenschmidt/build-ng-packagr`
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: A [custom CLI builder]( https://github.com/linnenschmidt/build-ng-packagr) is available that will help you copy the assets folder created by the bundler into your library package.  
_(is that what it does?)_

File: aio/content/guide/library-asset-guide.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+```
+
+once provided we can add below script in scripts:
+
+```sh
+"build-css" : "scss-bundle -p <project-path-to-scss-bundle.config.json>"
+```
+
+## Copying assets using custom builder
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: don't need a section for this either

File: aio/content/guide/library-asset-guide.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        "rootDir": "./tests/cases/simple/",
+        "outFile": "./bundled.scss",
+        "ignoreImports": ["~@angular/.*"],
+        "logLevel": "silent"
+    }
+}
+```
+
+once provided we can add below script in scripts:
+
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: When you have created the configuration file, add a script to invoke the tool.

_(where do you add this script?)_

File: aio/content/guide/library-asset-guide.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+```sh
+"build-css" : "scss-bundle -p <project-path-to-scss-bundle.config.json>"
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: _use example-code tag and identify the file to which you add this script:_
`<code-example language="bash" header="??name of file??">`

File: aio/content/guide/library-asset-guide.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+```
```
Comment (jbogarthyde) [inline aio/content/guide/library-asset-guide.md]: `</code-example>`

**Review by jbogarthyde — CHANGES_REQUESTED** [review]
I think this whole thing should probably be a section in "Creating Libraries", rather than a separate page.

If you like, I can  open another PR and use this as source to do that -- I think it will need some additional information, but this has given me a better idea of how to find it. 😃

**Comment by santoshyadav198613** [comment]
Hi @jbogarthyde ,
Yes, you can open a PR glad it helped to get started, should I make changes and push or you want to take it ahead from here?

**Comment by Splaktar** [comment]
Agreed that it should be added to the existing guide rather than as a separate guide.

**Comment by santoshyadav198613** [comment]
Cool I will make the changes.

**Comment by jbogarthyde** [comment]
Thanks @santoshyadav198613 👍 
Closing this in favor of  https://github.com/angular/angular/pull/34129

**Comment by Splaktar** [comment]
@jbogarthyde the link that you posted just goes to this same PR. Can you please post the link to your new PR?

**Comment by Splaktar** [comment]
Oh I found it, https://github.com/angular/angular/pull/34217.

**Pull request closed without merging**